Deprecate and cleanup `OC.L10N` in favor of `@nextcloud/l10n`

susnux opened the pull request:
## Summary
This PR consists of three parts:
1. Cleanup long standing deprecation from `OC`: `OC.addTranslations` was deprecated with NC17 and is not used anymore by any app within the github nextcloud organization
2. Deprecate `OC.getLocale` and `OC.getLanguage` as they are provided by `@nextcloud/l10n` and drop duplicated code
3. Deprecate and replace the translation functions implementation with the `@nextcloud/l10n`.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated
  - See PR: https://github.com/nextcloud/documentation/pull/9575 (upgrade guide)

susnux committed:
Drop `OC.addTranslations` deprecated since NC17 and not used by any NC app

Signed-off-by: Ferdinand Thiessen <rpm@fthiessen.de>

susnux committed:
Deprecate OC.L10N functions in favor of translation functions from `@nextcloud/l10n`

Signed-off-by: Ferdinand Thiessen <rpm@fthiessen.de>

ChristophWurst reviewed (approved):
:broom:

skjnldsv commented on core/src/OC/l10n.js:
```diff
@@ -28,79 +27,56 @@
 	/**
 	 * @private
+	 * @deprecated 26.0.0 use `_unregister` from https://www.npmjs.com/package/@nextcloud/l10n
 	 */
-	_unregister: unregisterAppTranslations,
+	_unregister,
```
is it a private method? Then it should not be exposed or it should be made public?

susnux replied:
This is currently only used in CI tests, but I do not see a reason why this function should not be on the public API.
Maybe we should change this on the `@nextcloud/l10n` package.

But for this PR I wanted to keep the changes small and do not introduce a breaking API change.

susnux commented:
/compile

susnux commented:
Rebased onto current master

susnux commented:
Blocked by nextcloud/nextcloud-l10n#571 (and we need to wait for a new release)
There was an issue introduced in the standalone implementation which caused the function to synchronously load translations instead of asynchronously, this is the reason the test fail here.

susnux committed:
fix(jsunit): Modify `OC.L10N` tests as it does not depend on `OC.` anymore

Meaning we can not stub `getLocale` but must override the document attribute instead.

Signed-off-by: Ferdinand Thiessen <rpm@fthiessen.de>

susnux commented:
/compile

susnux commented:
/compile

nextcloud-command committed:
Compile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

susnux commented:
The drone CI failure on the mysql DB tests seem unrelated.
